include/envoy/buffer/buffer.h

```diff
@@ -497,7 +490,7 @@ class Reservation final {
    * @note No other methods should be called on the object after `commit()` is called.
    */
   void commit(uint64_t length) {
-    ASSERT(length <= length_);
+    ENVOY_BUG(length <= length_, "commit() length must be <= size of the Reservation");
```

Comment:
Is the situation here that it's safe to have `length > length_`? Trying to understand why one condition is `ENVOY_BUG` and the other is `ASSERT`.

Reply:
Both conditions are the same, `length` must be less than or equal to `length_`, the change just moved it from ASSERT to ENVOY_BUG. 
(Maybe the semantics of ENVOY_BUG are confusing? ENVOY_BUG(condition, msg) is an error if condition was false, just like ASSERT(condition)).

Reply:
I made it ENVOY_BUG instead of ASSERT because this condition MAY mean that the caller thought they were committing more bytes than they actually commit, which MAY mean the data in the buffer is now not what the caller expected (a pretty serious bug). However, `commit()` will keep the data structure in a sane/safe state, so RELEASE_ASSERT seemed over-zealous.

Reply:
The bytes committed is min(length, length_).  Tests verify that this case is handled gracefully.

Reply:
Got it, thanks for the followup. Just trying to be clear on the use of `ASSERT` and `ENVOY_BUG` in this situation.